modules/multi2vec-clip/vectorizer/vectorizer_test.go

```diff
@@ -172,6 +174,31 @@ func TestVectorizerWithDiff(t *testing.T) {
 	}
 }
 
+func TestVectorizer_withWeights(t *testing.T) {
+	client := &fakeClient{}
+	vectorizer := &Vectorizer{client}
+	config := newConfigBuilder().
+		addSetting("imageFields", []interface{}{"image"}).
+		addSetting("textFields", []interface{}{"text"}).
+		addWeights([]interface{}{0.4}, []interface{}{0.6}).
+		build()
+	settings := NewClassSettings(config)
+
+	input := &models.Object{
+		ID: "some-uuid",
+		Properties: map[string]interface{}{
+			"image":       image,
+			"text":        "text",
+			"description": "non-vectorizable",
+		},
+	}
+
+	err := vectorizer.Object(context.Background(), input, nil, settings)
```

Comment:
where do these numbers come from? I assume it is 0.4*something1 + 0.6*something2

Reply:
They come from `fakes_for_test.go` and attempting to define the comparison programmatically walks into floating point comparison errors. E.g., comparing `16` with `16.002`, so I gave that up and just did it manually.

With the first entry of L197 as an example, it's calculated as:
```
(textVectors[0][0]*0.4+imageVectors[0][0]*0.6)/2 = (0.4 + 6) / 2 = 3.2
```

When weights aren't provided, as in L167, then the vectors are weighted by unity:
```
(textVectors[0][0]*1+imageVectors[0][0]*1)/2 = (1 + 10) / 2 = 5.5
```

This may or may not be expected by the user. I think I'd expect that providing no weights at all would weight each vector by `0.5` rather than `1` but it's a question of convention and precedence

Reply:
oki, could you add a short comment - I assume that your print came from asking yourself where the results are coming from? :)